tools/appveyor.psm1

```diff
@@ -3,14 +3,81 @@ $repoRoot = Join-Path $PSScriptRoot '..'
 
 Import-Module (Join-Path $repoRoot 'build.psm1')
 
+# tests if we should run a daily build
+# returns true if the build is scheduled 
+# or is a pushed tag
+Function Test-DailyBuild
+{
+    if($env:APPVEYOR_SCHEDULED_BUILD -eq 'True')
+    {
+        return $true
+    }
+    if($env:APPVEYOR_REPO_TAG_NAME)
+    {
+        return $true
+    }
+    return $false
+}
+
+# Sets a build variable
+Function Set-BuildVariable
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
```

Comment:
When `Mandatory=$true` is specified, `ValidateNotNullOrEmpty` is not necessary, because the former requires the value to be not null or empty.